File: Serene_Movies_Task/Serene_Movies_Task.Web/Modules/Default/Movie/MovieColumns.cs
using MovieTutorial;
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Serene_Movies_Task.Default.Columns;

[ColumnsScript("Default.Movie")]
[BasedOnRow(typeof(MovieRow), CheckNames = true)]
public class MovieColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int MovieId { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string Description { get; set; }
    public string Storyline { get; set; }
    public int Year { get; set; }
    public DateTime ReleaseDate { get; set; }
    public int Runtime { get; set; }
    [Width(200), GenreListFormatter , QuickFilter]
    public List<int> GenreList { get; set; }
    public MovieKind Kind { get; set; }
}

File: Serene_Movies_Task/Serene_Movies_Task.Web/Modules/ServerTypes/Default/MovieColumns.ts
import { GenreListFormatter } from "@/Default/Movie/GenreListFormatter";
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MovieKind } from "./MovieKind";
import { MovieRow } from "./MovieRow";

export interface MovieColumns {
    MovieId: Column<MovieRow>;
    Title: Column<MovieRow>;
    Description: Column<MovieRow>;
    Storyline: Column<MovieRow>;
    Year: Column<MovieRow>;
    ReleaseDate: Column<MovieRow>;
    Runtime: Column<MovieRow>;
    GenreList: Column<MovieRow>;
    Kind: Column<MovieRow>;
}

export class MovieColumns extends ColumnsBase<MovieRow> {
    static readonly columnsKey = 'Default.Movie';
    static readonly Fields = fieldsProxy<MovieColumns>();
}

[GenreListFormatter, MovieKind]; // referenced types